api4/post.go

```diff
@@ -581,6 +581,19 @@ func getPostThread(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Err = err
 		return
 	}
+	if list.FirstInaccessiblePostTime != 0 {
+		clientPostList := c.App.PreparePostListForClient(c.AppContext, list)
+		clientPostList, err = c.App.SanitizePostListMetadataForUser(c.AppContext, clientPostList, c.AppContext.Session().UserId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		if err := clientPostList.EncodeJSON(w); err != nil {
+			mlog.Warn("Error while writing response", mlog.Err(err))
+		}
+		return
```

Comment:
Why return from here? Ain't rest of the operations needed?
Like this one `if _, err = c.App.GetPostIfAuthorized(c.AppContext, post.Id, c.AppContext.Session(), false);` or this one `w.Header().Set(model.HeaderEtagServer, clientPostList.Etag())`?

Reply:
> The case I have in mind was for when the post isn't in the post list

We can't know that with `if list.FirstInaccessiblePostTime != 0 {`. We must be in the route you mentioned `post, ok := list.Posts[c.Params.PostId]` for that.

> The user will be seeing an error page and not a partial listing of threads

So, if list.FirstInaccessiblePostTime is > 0, then you can simply send empty posts list. And there won't be a need to call these functions: `PreparePostListForClient`, `SanitizePostListMetadataForUser`.

Reply:
> The case I have in mind was for when the post isn't in the post list

I thought you were talking about `c.Params.PostId` post, that's why I said it can be verified only with this `ok := list.Posts[c.Params.PostId]`.

But here `c.Params.PostId` doesn't have to be the _root post_. It can be _any_ post in a thread.
And as I see it now, the idea is to return empty list in case the RootPost is inaccessible and that would mean, if _any_ post is inaccessible, the RootPost would also be inaccessible.

So, I think your original top-level "if" check was correct to verify if _any_ post is inaccessible.

Because in current case, it may return partial-thread as long as `c.Params.PostId` is accessible.

Reply:
Ahh, makes sense. Moved the check back to the top level, and send empty list in that case.